### DISCUSSION TITLE
High memory usage (9.5 GB) and long scan time (45 min) on some repos

### DISCUSSION THREAD

**Discussion by ptupitsyn** [aquasecurity/trivy#6549] [Bugs]
### Description

Some repos, like https://github.com/kubernetes/minikube, take a very long time to scan (45 minutes on `t3.xlarge`) and consume up to 9.5 GB of RAM.

### Desired Behavior

Memory consumption below 1 GB, scan time under 5 minutes.

### Actual Behavior

Memory consumption of 9.5 GB, scan time 45 minutes.

### Reproduction Steps

```bash
1. git clone https://github.com/kubernetes/minikube.git
2. cd minikube
3. docker run -v $PWD:/myapp --entrypoint "trivy" aquasec/trivy --timeout 60m --quiet filesystem --scanners vuln,config --format json  /myapp
```

### Target

Filesystem

### Scanner

Vulnerability

### Output Format

JSON

### Mode

Standalone

### Debug Output

```bash
No output.
```

### Operating System

Ubuntu 22.04

### Version

```bash
0.50.2
```

### Checklist

- [X] Run `trivy image --reset`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
@DmitriyLewen Can you please take a look?

**Comment by simar7** [comment]
This might be related https://github.com/aquasecurity/trivy/discussions/6517

@ptupitsyn do you see the same behavior if you run the following:

Only `vuln` scanner
```docker run -v $PWD:/myapp --entrypoint "trivy" aquasec/trivy --timeout 60m --quiet filesystem --scanners vuln--format json  /myapp```

Only `misconfig` scanner
```docker run -v $PWD:/myapp --entrypoint "trivy" aquasec/trivy --timeout 60m --quiet filesystem --scanners config --format json  /myapp```

**Reply by simar7** [reply]
In my run, I only see a long running time with the misconfig scanner. I'll create an issue to track this.

**Reply by simar7** [reply]
Track: https://github.com/aquasecurity/trivy/issues/6557

**Reply by ptupitsyn** [reply]
Confirmed - `vuln` only completes in a few seconds.

**Comment by DmitriyLewen** [comment]
Related Issue - https://github.com/aquasecurity/trivy/issues/6557
